### PR TITLE
Separate code for computing input stocks

### PR DESCRIPTION
**Pull request by SallyDa**
Linked to [this sodym pr](https://github.com/JakobBD/sodym/pull/20)

**Update** (30th August)
* new directory structure, closing #1
* builds on changes to sodym to have MFASystem inherit from pydantic.BaseModel
* additional layer of code: PlasticsModel holds the definition, handles data reading and writing, and creates two different MFA systems, one for the historic part and another for the future part (the historic one is currently a bit unnecessary, but hopefully useful to have if we later make a more complex historic calculation as in the steel model)
* same idea for the steel model, with placeholder notes added for @Merjo 's updates. Note that this breaks the current steel model

### CODE REVIEW

File: src/model_extensions/use_stock_getter.py
Comment (JakobBD) [inline src/model_extensions/use_stock_getter.py:12]: As discussed with Merlin, I'm afraid that has to go back into the MFASystem namespace for the steel model. Sorry about the late notice!

**Review by JakobBD — CHANGES_REQUESTED** [review]

**Review by JakobBD — APPROVED** [review]

**Comment by JakobBD** [comment]
I approved the PR to not block merging for the whole week, but please change this before merging!

File: src/model_extensions/use_stock_getter.py
Comment (SallyDa) [reply src/model_extensions/use_stock_getter.py:12]: This is possibly the best argument for why we should initialise (or have the option to initialise) MFASystem directly with dims, processes, parameters, etc.

File: simson/common/data_transformations.py
Comment (JakobBD) [inline simson/common/data_transformations.py:64]: This should be in a separate file. There will be more functions with the same task but different methods. 

And so should extrapolate_stock.

Comment (SallyDa) [reply simson/common/data_transformations.py:64]: Okay, I don't think this should prevent the pr being merged, as we can separate the functions later when we add the extra cases (and if we have multiple options for the same task it might make sense to have a class structure where we can specify input and output). Did you manage to go through the rest of the pr, are you otherwise happy?

**Comment by JakobBD** [comment]
I've had a look yesterday. Certainly nothing that would stop me from merging now. Looks good!

**Pull request merged**
